service-tests/gem.js

```diff
@@ -0,0 +1,197 @@
+'use strict';
+
+const Joi = require('joi');
+const ServiceTester = require('./runner/service-tester');
+
+const {
+  isVPlusDottedVersionAtLeastOne,
+  isMetric
+} = require('./helpers/validators');
+const isOrdinalNumber = Joi.string().regex(/^[0-9]+(ᵗʰ|ˢᵗ|ⁿᵈ|ʳᵈ)$/);
+const isOrdinalNumberDaily = Joi.string().regex(/^[0-9]+(ᵗʰ|ˢᵗ|ⁿᵈ|ʳᵈ) daily$/);
+
+const t = new ServiceTester({ id: 'gem', title: 'Ruby Gems' });
+module.exports = t;
+
+
+// version endpoint
+
+t.create('version (valid)')
+  .get('/v/formatador.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'gem',
+    value: isVPlusDottedVersionAtLeastOne
+  }));
+
+t.create('version (not found)')
+  .get('/v/not-a-package.json')
+  .expectJSON({name: 'gem', value: 'not found'});
+
+t.create('version (connection error)')
+  .get('/v/formatador.json')
+  .networkOff()
+  .expectJSON({name: 'gem', value: 'inaccessible'});
+
+t.create('version (unexpected response)')
+  .get('/v/formatador.json')
+  .intercept(nock => nock('https://rubygems.org')
+    .get('/api/v1/gems/formatador.json')
+    .reply(200, "{{{{{invalid json}}")
+  )
+  .expectJSON({name: 'gem', value: 'invalid'});
+
+
+// downloads endpoints
+
+// total downloads
+t.create('total downloads (valid)')
+  .get('/dt/rails.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'downloads',
+    value: isMetric
+  }));
+
+t.create('total downloads (not found)')
+  .get('/dt/not-a-package.json')
+  .expectJSON({name: 'downloads', value: 'not found'});
+
+t.create('total downloads (connection error)')
+  .get('/dt/rails.json')
+  .networkOff()
+  .expectJSON({name: 'downloads', value: 'inaccessible'});
+
+t.create('total downloads (unexpected response)')
+  .get('/dt/rails.json')
+  .intercept(nock => nock('https://rubygems.org')
+    .get('/api/v1/gems/rails.json')
+    .reply(200, "{{{{{invalid json}}")
+  )
+  .expectJSON({name: 'downloads', value: 'invalid'});
+
+
+// version downloads
+t.create('version downloads (valid, stable version)')
+  .get('/dv/rails/stable.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'downloads',
+    value: Joi.string().regex(/^[1-9][0-9]*[kMGTPEZY]? stable version$/)
+  }));
+
+t.create('version downloads (valid, specific version)')
+  .get('/dv/rails/4.1.0.json')
+  .expectJSONTypes(Joi.object().keys({
+    name: 'downloads',
+    value: Joi.string().regex(/^[1-9][0-9]*[kMGTPEZY]? version 4.1.0$/)
```

Comment:
I think these should be changed to **downloads@4.1.0** and **downloads@stable**. That’s similar to the **npm@next** badge. Would you be up for changing them?